pkg/service/providercacher/cachingqueue.go

```diff
@@ -8,15 +8,35 @@ import (
 )
 
 type (
+	CachingQueueQueuer interface {
+		Queue(ctx context.Context, job ProviderCachingJob) error
+	}
+
+	CachingQueueReader interface {
+		Read(ctx context.Context, maxJobs int) ([]ProviderCachingJob, error)
+	}
+
+	CachingQueueReleaser interface {
+		Release(ctx context.Context, jobID string) error
+	}
+
+	CachingQueueDeleter interface {
+		Delete(ctx context.Context, jobID string) error
+	}
```

Comment:
I would argue that the reader interface should include release and delete functions...it is maybe a "consumer".

Reply:
the convention in go is adding an "er" suffix to one-method interfaces. I find that a bit confusing, to be honest, because the "reader" will be the one calling the "read" method, and not the one implementing it, which would be a "readable". Therefore, if I were to put all these together in a single interface I would call it "consumable" instead. More probably, I'd just collapse all the one-method interfaces in the single `CachingQueue` interface and call it a day.

I'll leave it as is for now if you don't mind. I don't think it's hard to read and understand as it is and we will likely not need to touch it too much in the future anyway.